core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -1537,13 +1540,16 @@ FileAggregatedMultipart aggregatedMultipart() {
         QueryParams queryParams() {
             QueryParams result = queryParams;
             if (result == null) {
-                synchronized (this) {
+                lock.lock();
+                try {
                     result = queryParams;
```

Comment:
What do you think of removing `lock` from this method. It's likely that the method is safe on call multiple invocations and it would be rare for multiple threads to access that.